### PR TITLE
Adddsytd2

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by btracey** [comment]
This fails because it needs https://github.com/gonum/blas/pull/148

File: native/dsytd2.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	}
+	if len(tau) < n-1 {
+		panic(badTau)
+	}
+	if n <= 0 {
+		return
+	}
+	bi := blas64.Implementation()
+	if uplo == blas.Upper {
+		// Reduce the upper triangle of A
```
Comment (kortschak) [inline native/dsytd2.go]: Periods at sentence ends.

File: native/dsytd2.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+				bi.Dsyr2(uplo, i+1, -1, a[i+1:], lda, tau, 1, a, lda)
+				a[i*lda+i+1] = e[i]
+			}
+			d[i+1] = a[(i+1)*lda+i+1]
+			tau[i] = taui
+		}
+		d[0] = a[0]
+		return
+	}
+	for i := 0; i < n-1; i++ {
```
Comment (kortschak) [inline native/dsytd2.go]: Comment missing above this loop.

File: native/dsytd2.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+		// Generate elementary reflector H[i] = I - tau * v * v^T to
+		// annihilate A[i+2:n, i]
+		var taui float64
+		a[(i+1)*lda+i], taui = impl.Dlarfg(n-i-1, a[(i+1)*lda+i], a[min(i+2, n-1)*lda+i:], lda)
+		e[i] = a[(i+1)*lda+i]
+		if taui != 0 {
+			a[(i+1)*lda+i] = 1
```
Comment (kortschak) [inline native/dsytd2.go]: Comment missing above this assignment.

File: native/dsytd2.go (hunk 1 above)
Comment (btracey) [inline native/dsytd2.go]: Done.

File: native/dsytd2.go (hunk 2 above)
Comment (btracey) [inline native/dsytd2.go]: Done.

File: native/dsytd2.go (hunk 3 above)
Comment (btracey) [inline native/dsytd2.go]: Done.

**Comment by btracey** [comment]
PTAL

**Comment by kortschak** [comment]
LGTM

**Pull request merged**